app/pages/project/instances/instance/tabs/SettingsTab.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { format, formatDistanceToNow } from 'date-fns'
+import { useEffect, useId, useState, type ReactNode } from 'react'
+import { useForm } from 'react-hook-form'
+
+import { apiQueryClient, useApiMutation, usePrefetchedApiQuery } from '~/api'
+import { ListboxField } from '~/components/form/fields/ListboxField'
+import { useInstanceSelector } from '~/hooks/use-params'
+import { addToast } from '~/stores/toast'
+import { Button } from '~/ui/lib/Button'
+import { FieldLabel } from '~/ui/lib/FieldLabel'
+import { LearnMore, SettingsGroup } from '~/ui/lib/SettingsGroup'
+import { TipIcon } from '~/ui/lib/TipIcon'
+import { links } from '~/util/links'
+
+Component.displayName = 'SettingsTab'
+export function Component() {
+  const instanceSelector = useInstanceSelector()
+
+  const [now, setNow] = useState(new Date())
+
+  useEffect(() => {
+    const timer = setInterval(() => setNow(new Date()), 1000)
+    return () => clearInterval(timer)
+  }, [])
+
+  const { data: instance } = usePrefetchedApiQuery('instanceView', {
+    path: { instance: instanceSelector.instance },
+    query: { project: instanceSelector.project },
+  })
+
+  const instanceUpdate = useApiMutation('instanceUpdate', {
+    onSuccess() {
+      apiQueryClient.invalidateQueries('instanceView')
+      addToast({ content: 'Instance auto-restart policy updated' })
+    },
+  })
+
+  const form = useForm({
+    defaultValues: {
+      autoRestartPolicy: instance.autoRestartPolicy,
+    },
+  })
+  const { isDirty } = form.formState
+
+  const onSubmit = form.handleSubmit((values) => {
+    instanceUpdate.mutate({
+      path: { instance: instanceSelector.instance },
+      query: { project: instanceSelector.project },
+      body: {
+        ncpus: instance.ncpus,
+        memory: instance.memory,
+        bootDisk: instance.bootDiskId,
```

Comment:
This strikes me as an awkward API design. Let's say the user just wants to update the `autoRestartPolicy` and does not include the `bootDisk`, it would then unset the instances boot disk. It also makes updating this in the CLI cumbersome.

Obviously it's outside of the scope of this PR, but thought I'd mention.

Reply:
We've talked about this a lot on the API side (see https://github.com/oxidecomputer/omicron/issues/6406). Most of our PUTs don't work this way, but it makes things simpler and more consistent if they do. Having to specify all the fields every time is of course a downside for clients. The most basic problem with making the fields optional is that if leaving a field out doesn't unset, you still need to be able to unset one if you want, which means you have to have different semantics for "`bootDisk` not present" and `bootDisk: null`, and that's both confusing and not spec compliant.

Reply:
And I suppose having dedicated instance resize and update instance restart policy endpoints adds a bunch of extra code on the API? I was think less optionality and more something that would facilitate (CLI) `instance auto-restart policy enable`.

Reply:
Yeah, we don't really want to proliferate endpoints like that. My inclination would be to write that command by hand on the CLI side without it being a dedicated endpoint, or better yet come up with a pattern (a flag to `instance update`?) that lets you update one field while telling it to handle the rest for you.